crowbar_framework/app/models/service_object.rb

```diff
@@ -257,20 +257,31 @@ def remove_pending_elements(bc, inst, elements)
     end
   end
 
+  def restore_node(node)
+    node.crowbar['state'] = 'ready'
+    node.crowbar['state_owner'] = ""
+    node.save
+  end
+
   def restore_to_ready(nodes)
     with_lock "BA-LOCK" do
       nodes.each do |node_name|
         node = NodeObject.find_node_by_name(node_name)
-        next if node.nil?
-
-        # Nothing to delay so mark them applying.
-        node.crowbar['state'] = 'ready'
-        node.crowbar['state_owner'] = ""
-        node.save
+        next if node.nil? or node.crowbar['state'] == 'upgrade'
+          # Nothing to delay so mark them applying.
+          restore_node(node)
       end
     end
   end
 
+  def remove_upgrade_state(nodes)
```

Comment:
This is never called so far. i thought about adding a button in the node view to reset the node state.  Resetting the upgrade state will remove a few more steps too, as they don't have the chef-client running.